app/Http/Controllers/Table/PortNacController.php

```diff
@@ -72,10 +73,16 @@ protected function sortFields($request)
      */
     public function baseQuery($request)
     {
-        return PortsNac::select('port_id', 'mac_address', 'ip_address', 'vlan', 'domain', 'host_mode', 'username', 'authz_by', 'timeout', 'time_elapsed', 'time_left', 'authc_status', 'authz_status', 'method')
-            ->where('device_id', $request->device_id)
-            ->hasAccess($request->user())
-            ->with('port');
+        if (is_numeric($request->device_id)) {
+            return PortsNac::select('device_id', 'port_id', 'mac_address', 'ip_address', 'vlan', 'domain', 'host_mode', 'username', 'authz_by', 'timeout', 'time_elapsed', 'time_left', 'authc_status', 'authz_status', 'method')
```

Comment:
Convert this to one query to prevent divergence between the queries using ->when()

Reply:
done